src/theory/quantifiers/sygus/sygus_unif_rl.cpp

```diff
@@ -563,6 +563,17 @@ Node SygusUnifRl::DecisionTreeInfo::buildSolAllCond(Node cons,
   // add conditions
   d_conds.clear();
   d_conds.insert(d_conds.end(), d_cond_mvs.begin(), d_cond_mvs.end());
+  // shuffle conditions before bulding DT
+  //
+  // this does not impact whether it's possible to build a solution, but it does
+  // impact the potential size of the resulting solution (can make it smaller,
+  // bigger, or have no impact) and which conditions will be present in the DT,
+  // which influences the "quality" of the solution for cases not covered in the
+  // current data points
+  if (options::sygusUnifShuffleCond())
+  {
+    std::shuffle(d_conds.begin(), d_conds.end(), d_rng);
```

Comment:
Use std::random_shuffle.  I haven't seen this d_rng thing before. 

Here's an example of shuffling:
https://github.com/CVC4/CVC4/blob/master/src/theory/quantifiers/cegqi/ceg_bv_instantiator.cpp#L309

Reply:
I saw online that `std::shuffle` was preferable to `std::random_shuffle` from C++11 on. And I'm parameterizing it with a random generator so that we can have control over the sequence. This is necessary for reproducing results.

Reply:
I'm just following what we did for the CAV 2018 BV algorithm. Perhaps @aniemetz can weigh in?

Reply:
The approach @HanielB proposes is actually the way to go. We want any randomness in CVC4 to be deterministic, hence the existence of d_rng. The example @ajreynol posted should be changed accordingly. It seems that I overlooked this back then when I introduced the random number generator in #1370.

Reply:
I just realized that @HanielB is not using CVC4's RNG. He should.

Reply:
I'll submit a PR for the change in ceq_bv_instantiator.cpp.